packages/dd-trace/src/telemetry/endpoints.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+
+const dc = require('dc-polyfill')
+const { sendData } = require('./send-data')
+
+const fastifyRouteCh = dc.channel('apm:fastify:route:added')
+
+let config
+let application
+let host
+let getRetryData
+let updateRetryData
+
+/**
+ * Keep track of endpoints that still need to be sent.
+ * Map key is `${METHOD} ${PATH}`, value is { method, path }
+ */
+const pendingEndpoints = new Map()
+let flushScheduled = false
+let isFirstPayload = true
+
+function endpointKey (method, path) {
+  return `${method} ${path}`
+}
+
+function scheduleFlush () {
+  if (flushScheduled) return
+  flushScheduled = true
+  setImmediate(flushAndSend).unref()
+}
+
+function recordEndpoint (method, path) {
+  if (!method || !path) return
```

Comment:
not sure this is necesary, you already check path before, and not sure how method could be empty